### PR TITLE
Move the fieldspec base vector.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

File: searchlib/src/tests/queryeval/blueprint/blueprint_test.cpp [hunk 1]
```diff
@@ -808,26 +766,13 @@ Test::requireThatDocIdLimitInjectionWorks()
...
+    EXPECT_EQUAL(40u, sizeof(Blueprint::State));
+    EXPECT_EQUAL(32u, sizeof(Blueprint));
+    EXPECT_EQUAL(72u, sizeof(LeafBlueprint));
 }
 
-TEST_APPHOOK(Test);
+TEST_MAIN() {
+    TEST_DEBUG("lhs.out", "rhs.out");
+    TEST_RUN_ALL();
+}
```
Comment (havardpe) [inline searchlib/src/tests/queryeval/blueprint/blueprint_test.cpp:778]: please add newline

**Review by havardpe — DISMISSED** [review]

**Review by havardpe — COMMENTED** [review]
are we sure we never get > 64k query nodes (max tree size)?

File: searchlib/src/tests/queryeval/blueprint/blueprint_test.cpp [hunk 2]
```diff
@@ -149,49 +149,29 @@ struct MyTerm : SimpleLeafBlueprint {
...
-    void testChildOrCollapsing();
-    void testSearchCreation();
-    void testBlueprintMakeNew();
-    void requireThatAsStringWorks();
-    void requireThatAsSlimeWorks();
-    void requireThatVisitMembersWorks();
-    void requireThatDocIdLimitInjectionWorks();
-    int Main() override;
+    static bool check_equal(const SearchIterator &a, const SearchIterator &b);
+    bool check_not_equal(const SearchIterator &a, const SearchIterator &b);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/tests/queryeval/blueprint/blueprint_test.cpp]: Fixed

File: searchlib/src/vespa/searchlib/queryeval/field_spec.h [hunk 3]
```diff
@@ -60,7 +60,14 @@ class FieldSpecBaseList
     List _list;
 
 public:
+    FieldSpecBaseList() = default;
+    FieldSpecBaseList(FieldSpecBase spec) : _list() { _list.push_back(spec); }
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/field_spec.h]: Removed

**Pull request merged**